src/llvm-pass-helpers.cpp

```diff
@@ -124,6 +125,12 @@ namespace jl_intrinsics {
     // The allocation size is set to the first argument.
     static Function *addGCAllocAttributes(Function *target)
     {
+        auto FnAttrs = AttrBuilder(target->getContext());
+        FnAttrs.addAllocKindAttr(AllocFnKind::Alloc | AllocFnKind::Uninitialized);
```

Comment:
I don't think `Uninitialized` is strictly true enough to mark this. It is partially initialized, and in particular, is partially zero'd so that reading some fields is not UB, but reading other fields is UB.

Reply:
So that's what's annoying, because we pass a pointer to the middle of the object, though I'm not sure if this matters that much so we can change it.

Reply:
yeah, it is annoying that we cannot mark it better, but I think the GEP itself is supposed to add this info based on the inbounds flag, so we should be able to expect LLVM is supposed to be able to handle that.

Reply:
Actually, thinking about it. It is unitialized, we only store the tag in LLVM.

Reply:
Ah right, we do the initialization to make it a valid allocation inside LLVM code itself afterwards